### ISSUE TITLE
"$ tensorflow_model_server --version" returned "Illegal instruction"

### ISSUE THREAD

**Issue by liujianjunhq** [clearlinux/distribution#1238]
After bundle-add tensorflow-serving, it reported "Illegal instruction" when running "Illegal instruction". 
root@44cfcb6113b9/ # swupd bundle-add tensorflow-serving
Loading required manifests...
No packs need to be downloaded
Starting download of remaining update content. This may take a while...
        ...100%
Installing bundle(s) files...
        ...100%
Calling post-update helper scripts
Successfully installed 1 bundle
root@44cfcb6113b9/ # tensorflow_model_server --version
Illegal instruction

**Comment by liujianjunhq** [comment]
As analyze via clr-avx-tools, tensorflow-serving included AVX-512 instructions as below. But my PC do not support AVX-512 instruction as checked via "cat /proc/cpuinfo". 
I could not reproduce this issue on another PC with AVX-512 supported. 
So, the issue should be caused by the AVX-512 instructions included. Need to update the tensorflow-serving build option to support the basic CPU instruction set. 

.../usr/share/clr-avx-tools # python3 avxjudge.py /usr/bin/tensorflow_model_server 
Analyzing /usr/bin/tensorflow_model_server
Top SSE functions by instruction count
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        9.57 %s
    mtx_init@@Base                      6.26 %s
    _ZNKSt5ctypeIcE8do_widenEc@@Base    4.36 %s
    malloc_hook                         2.65 %s
    text_env                            2.09 %s

Top SSE functions by value
    mtx_init@@Base                      174584.01
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        155001.33
    _ZNKSt5ctypeIcE8do_widenEc@@Base    77825.56
    text_env                            11413.65
    _ZNSt11logic_errorC1ERKS_@@Base     6333.84

Top AVX2 functions by instruction count
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        1.0 %s
    _ZNKSt5ctypeIcE8do_widenEc@@Base    0.98 %s
    text_env                            0.79 %s
    mtx_init@@Base                      0.65 %s
    _ZNSt15basic_stringbufIcSt11char_traitsIcESaIcEED1Ev@@Base  0.35 %s

Top AVX2 functions by value
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        35006.44
    mtx_init@@Base                      14002.99
    _ZNKSt5ctypeIcE8do_widenEc@@Base    12309.51
    text_env                            3807.98
    _ZNSt11logic_errorC1ERKS_@@Base     407.47

Top AVX512 functions by instruction count
    _ZNSt15basic_stringbufIcSt11char_traitsIcESaIcEED1Ev@@Base  2.95 %s
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        2.29 %s
    mtx_init@@Base                      1.37 %s
    text_env                            0.57 %s
    _ZNKSt5ctypeIcE8do_widenEc@@Base    0.45 %s

Top AVX512 functions by value
    _ZStplIcSt11char_traitsIcESaIcEESbIT_T0_T1_EPKS3_RKS6_@@Base        51071.6
    mtx_init@@Base                      37790.41
    _ZNKSt5ctypeIcE8do_widenEc@@Base    5174.55
    text_env                            2355.71
    _ZNSt15basic_stringbufIcSt11char_traitsIcESaIcEED1Ev@@Base  212.45

**Comment by bryteise** [comment]
Generally we have built 3 different libraries for tensorflow to support westmere, avx2 and avx512 instruction sets. I note this does have a impact on the size of package.

**Comment by thiagomacieira** [comment]
This is not a library. This is a binary in /usr/bin.

**Comment by bryteise** [comment]
Ah right though the same comment applies where we have executables for each target arch.

**Comment by thiagomacieira** [comment]
We do, but this is the one in /usr/bin, not /usr/bin/haswell/avx512_1

**Comment by liujianjunhq** [comment]
I checked that the same target file inside tensorflow/serving container (via run "docker pull tensorflow/serving")  could support AVX2 as default. 
I just checked in CLR, /usr/bin/tensorflow_model_server built with --config=release also supports AVX2 instructions. 

Working on updating the build option to remove all AVX2/AVX_512 instructions: 
I tried to build with "--copt=-O3 --copt=-march=westmere", but there are still AVX2 instructions left as below.
Top AVX2 functions by instruction count
    text_env                            0.56 %s

Top AVX2 functions by value
    text_env                            3005.35
